### PR TITLE
Add JSON format and timestamp for logs

### PR DESCRIPTION
**Pull request by axelfahy**
### Proposed changes

* Add the option to have a JSON format for the logs if the environment variable `OPENCTI_JSON_LOGGING` is set to True.  This adds a new dependency: `python_json_logger`.
* Add a timestamp, using a custom Formatter, in the logs to facilitate the reading.

Example of output: 

```
{"timestamp": "2021-09-15T09:34:43.106091Z", "level": "INFO", "name": "root", "message": "Listing Threat-Actors with filters null."}
{"timestamp": "2021-09-15T09:34:43.116402Z", "level": "DEBUG", "name": "urllib3.connectionpool", "message": "Starting new HTTP connection (1): opencti:8080"}
```

### Related issues

* #207 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality using different use cases
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

This is the current workaround I use to have the logs in JSON format, but maybe you have a cleaner solution or a better alternative to propose.

### CODE REVIEW

**Comment by nor3th** [comment]
Hey @axelfahy 

Thank you very much for your idea and the work you have invested, but since the OpenCTIApiClient isn't handling environment variables, I'd kindly ask you to implement following changes.

Reading the environment variable should be implemented here: https://github.com/OpenCTI-Platform/client-python/blob/8ae0fc866740e9d1f314988e39ebe9dad9f82656/pycti/connector/opencti_connector_helper.py#L383

Then add a log_type variable (with the default value False) to the OpenCTIApiClient constructor here https://github.com/OpenCTI-Platform/client-python/blob/8ae0fc866740e9d1f314988e39ebe9dad9f82656/pycti/api/opencti_api_client.py#L81

The logging implementation itself you can leave as it is. 

Please test the code by adding `os.environ["OPENCTI_JSON_LOGGING"] = true` to one of the test connectors here https://github.com/OpenCTI-Platform/client-python/blob/master/tests/cases/connectors.py

You can run the connector tests with `pytest -k test_internal_enrichment_connector --connectors -vvl` (those tests have to be run on your own OpenCTI instance with your own API key, since the RabbitMq connection doesn't work with demo.opencti.io)

If you have any questions, please don't hesitate to ask.

Thank you

File: requirements.txt [hunk 1]
```diff
@@ -5,5 +5,6 @@ datefinder~=0.7
 stix2~=2.1.0
 pika~=1.2
 sseclient~=0.0.27
+python_json_logger==2.0.2
```
Comment (nor3th) [inline requirements.txt:8]: Please change that to `python_json_logger~=2.0.2`

**Review by nor3th — CHANGES_REQUESTED** [review]
requested changes have already been added in an earlier comment

File: requirements.txt (hunk 1 above)
Comment (axelfahy) [reply requirements.txt]: Thank you for your feedbacks @nor3th, I have made the changes. 

However, I was not able to run the tests: `pytest -k test_internal_enrichment_connector --connectors -vvl`:
`tests/02-integration/connector/test_connector.py::test_internal_enrichment_connector[ipv4_enrichment]` hangs indefinitely, the message created by the connector is never consumed in RabbitMQ, do you have any idea why? It works fine with real connectors. 

I will do the appropriate changes in the worker as well (passing the `json_logging` variable to the constructor of the`OpenCTIApiClient` used by the worker).

File: pycti/api/opencti_api_client.py [hunk 2]
```diff
@@ -94,7 +117,18 @@ def __init__(self, url, token, log_level="info", ssl_verify=False, proxies={}):
         numeric_level = getattr(logging, self.log_level.upper(), None)
         if not isinstance(numeric_level, int):
             raise ValueError("Invalid log level: " + self.log_level)
-        logging.basicConfig(level=numeric_level)
+
+        logging.error(f"TESTS : {json_logging=}")
```
Comment (nor3th) [inline pycti/api/opencti_api_client.py:121]: Can you please remove this line. I think that's a left over from your testing phase

File: requirements.txt (hunk 1 above)
Comment (nor3th) [reply requirements.txt]: Thank you for the changes. I'll have a look at that the test case in the upcoming days if that's fine with you?

File: pycti/api/opencti_api_client.py (hunk 2 above)
Comment (axelfahy) [reply pycti/api/opencti_api_client.py]: Sorry about that, removed.

File: requirements.txt (hunk 1 above)
Comment (axelfahy) [reply requirements.txt]: Sure, thank you!

Comment (axelfahy) [reply requirements.txt]: The tests are working, it was an error with my local OpenCTI configuration.

Comment (nor3th) [reply requirements.txt]: Sounds great, thank you for your effort!

**Review by nor3th — APPROVED** [review]

**Pull request merged**